src/main/java/hudson/plugins/gradle/injection/InjectionConfig.java

```diff
@@ -241,21 +241,19 @@ public void setMavenInjectionDisabledNodes(List<NodeLabelItem> mavenInjectionDis
     }
 
     @DataBoundSetter
-    public void setInjectionVcsRepositoryPatterns(String injectionVcsRepositoryPatterns) {
-        this.injectionVcsRepositoryPatterns = Util.fixEmptyAndTrim(injectionVcsRepositoryPatterns);
+    public void setVcsRepositoryFilter(String vcsRepositoryFilter) {
+        this.vcsRepositoryFilter = Util.fixEmptyAndTrim(vcsRepositoryFilter);
 
-        this.parsedInjectionVcsRepositoryPatterns = this.injectionVcsRepositoryPatterns == null
-                ? ImmutableList.of()
-                : ImmutableList.copyOf(Arrays.stream(this.injectionVcsRepositoryPatterns.split(",")).map(String::trim).collect(Collectors.toList()));
+        this.parsedVcsRepositoryFilter = VcsRepositoryFilter.of(this.vcsRepositoryFilter);
```

Comment:
Let's maybe store the raw value inside the `VcsRepositoryFilter` class. This way we will avoid storing the same data twice in the config and it will guarantee that values are always in sync.

Reply:
Agree, will change

Reply:
Done